see/context/resources/qemu.py

```diff
@@ -369,18 +379,19 @@ def _retrieve_pool(self):
                 self.configuration['disk']['clone']['storage_pool_path'])
         else:
             return pool_lookup(self._hypervisor,
-                               self.configuration['disk']['image'])
+                               self.provider_image)
 
     def _retrieve_disk_path(self):
         if 'clone' in self.configuration['disk']:
-            return self._clone_disk(self.configuration['disk'])
+            return self._clone_disk(self.configuration['disk']['clone'],
+                                    self.provider_image)
         else:
-            return self.configuration['disk']['image']
+            return self.provider_image
 
-    def _clone_disk(self, configuration):
+    def _clone_disk(self, configuration, image):
```

Comment:
I guess you can drop the extra parameter and access to it directly with self.provider_image